docs/testplan/smart-switch/high-availability/keysight_ha_testplan.md

```diff
@@ -0,0 +1,408 @@
+* Mark time when the Standby DPU becomes Active and fully running traffic.
+* Using traffic generator tools to verify metrics associated with number of
+Concurrent Connections, Connection rate, and TCP/UDP failures collect data
+before, during and after switchover.
+
+### Test case3 Planned Switchover Perfect Sync Between HA Sets
+#### Test Objective 3
+* Reference HA-SmartSwitch-test_plan Module 1 Normal OP and Planned Events.
+  Reference the following test cases:
+    * Normal Op - Active
+    * Normal Op - Standby
+    * Planned Switchover - Active
+    * Planned Switchover - Backup
+
+![HA Perfect Sync Between HA Sets](images/ha_linkloss_test.svg)
+#### Steps for Test Case 3
+* Refer to the section Common Steps Configuration1.
+* Verify traffic is flowing without any loss.
+* Add a second SmartSwitch with DPU set to Standby.
+* Verify normal operation.  Active DPU and Standby DPU states remain the same.
+* Through mgmt switch send commands to sync the Standby DPU to the Active.
+the second DPU shall join the HA set.
+* Using traffic generator tools to verify metrics associated with number of
+  Concurrent Connections, Connection rate, and TCP/UDP failures collect data
+  before, during and after switchover.
+
+### Test case4 Link Failures Active NPU to DPU Probe Drop Active
+#### Test Objective 4
+Verify packet flow when Active NPU to DPU link starts dropping probe packets.
+* Reference HA-SmartSwitch-test_plan Module 4 Link Failures.
+
+![HA LinkLoss](images/ha_linkloss_test.svg)
+#### Steps for Test Case 4
+* Refer to the section Common Steps Configuration2.
+* Verify traffic is flowing without any loss through the Active side.
+* Through mgmt remove link connection between Active DPU and NPU to drop packets.
+* Add ACL to block traffic the state should not change.
+* Mark start time at beginning of test as link connection is removed.
+* Role of the DPUs will not change.
+* Standby DPU will start to receive traffic will be verified by rx/tx counters.
+* DPU of SmartSwitch0 stays active.
+* DPU of SmartSwitch1 remains standby.
+* Measure convergence time from start of the link removal between
+SmartSwitch0 and SmartSwitch1 switchover.
+* Using traffic generator tools to verify metrics associated with number of
+Concurrent Connections, Connection rate, and TCP/UDP failures collect data
+before, during and after unplanned event.
+
+### Test case5 Link Failures Active NPU to DPU Probe Drop Standby
+#### Test Objective 5
+Verify packet flow when Active NPU to DPU link starts dropping probe packets.
+* Reference HA-SmartSwitch-test_plan Module 4 Link Failures.
+
+![HA LinkLoss](images/ha_linkloss_test.svg)
+#### Steps for Test Case 5
+* Refer to the section Common Steps Configuration2.
+* Verify traffic is flowing without any loss through the Standby side.
+* Through mgmt remove link connection between Active DPU and NPU to drop packets.
+* Add ACL on DPU side to completely block the data port to simulate the DPU
```

Comment:
not completely block the data port, but block the probe traffic only by using destination port as condition.